sources/packages/backend/apps/api/src/route-controllers/restriction/restriction.controller.service.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  Param,
+  ParseIntPipe,
+  NotFoundException,
+  Injectable,
+} from "@nestjs/common";
+import { StudentRestrictionService } from "../../services";
+import {
+  RestrictionSummaryAPIOutDTO,
+  RestrictionDetailAPIOutDTO,
+} from "./models/restriction.dto";
+import { getUserFullName } from "../../utilities";
+import { ApiNotFoundResponse } from "@nestjs/swagger";
+import { RestrictionNotificationType } from "@sims/sims-db/entities";
+
+@Injectable()
+export class RestrictionControllerService {
+  constructor(
+    private readonly studentRestrictionService: StudentRestrictionService,
+  ) {}
+
+  /**
+   * Get restrictions for a student.
+   * @param studentId id of the student to retrieve restrictions.
+   * @param options query options.
+   * - `filterRestrictions` option to filter restrictions based on notificationType and restrictionCategory.
+   * @returns student restrictions for the provided student id.
+   */
+  async getStudentRestrictions(
+    @Param("studentId", ParseIntPipe) studentId: number,
+    options?: {
+      filterRestrictions?: boolean;
+    },
+  ): Promise<RestrictionSummaryAPIOutDTO[]> {
+    let studentRestrictions =
+      await this.studentRestrictionService.getStudentRestrictionsById(
+        studentId,
+      );
+    if (options?.filterRestrictions) {
+      const filteredStudentRestrictions = studentRestrictions.filter(
```

Comment:
The filter can be applied in the query, instead of filtering the data.

Reply:
To be clear, this will be a blocker.